### PR TITLE
Add missing spec URLs for API features

### PR DESCRIPTION
**Pull request by queengooborg**
This PR supersedes #12709.  This PR adds many new spec URLs to the API data based upon data from `@webref/idl`. 
 After rebasing #12709, I ran the script introduced in #16052 to fix the property sorting and move the spec URLs up in the JSON, then re-ran https://gist.github.com/gsnedders/0a475683482552a47e65d55df7bd5289 to add the missing spec URLs.

Notice: if there are any changes where other properties are moved around, they were not meant to be included this PR and can be reverted.  Reviewers with edit access to BCD are more than welcome to make changes directly to this PR.

I don't expect this PR to be merged in one go due to its sheer size.  Rather, I'm planning to separate this into multiple PRs and use this one as a breakdown to show what changes remain.

### CODE REVIEW

**Comment by teoli2003** [comment]
@queengooborg Should we close this bug, while still using it as a kind of tracker? (I know that the risk of an erroneous merge is low, but…)

**Comment by queengooborg** [comment]
I haven't finished splitting out the individual changes so this PR still has some changes to go.  I don't think we should close this PR as a result, but I'm down for marking it as a draft to imply not to merge this PR!

**Comment by queengooborg** [comment]
All of the other PRs have been merged, so this one is now ready for review!

File: api/ByteLengthQueuingStrategy.json [hunk 1]
```diff
@@ -116,6 +116,7 @@
       },
       "highWaterMark": {
         "__compat": {
+          "spec_url": "https://streams.spec.whatwg.org/#blqs-high-water-mark",
```
Comment (sideshowbarker) [inline api/ByteLengthQueuingStrategy.json:119]:
```suggestion
          "spec_url": "https://streams.spec.whatwg.org/#ref-for-blqs-high-water-mark①",
```

File: api/MutationRecord.json [hunk 2]
```diff
@@ -99,6 +100,7 @@
       "attributeName": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/MutationRecord/attributeName",
+          "spec_url": "https://dom.spec.whatwg.org/#dom-mutationrecord-attributename",
```
Comment (sideshowbarker) [inline api/MutationRecord.json:103]:
```suggestion
          "spec_url": "https://dom.spec.whatwg.org/#ref-for-dom-mutationrecord-attributename②",
```

File: api/MutationRecord.json [hunk 3]
```diff
@@ -147,6 +149,7 @@
       "attributeNamespace": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/MutationRecord/attributeNamespace",
+          "spec_url": "https://dom.spec.whatwg.org/#dom-mutationrecord-attributenamespace",
```
Comment (sideshowbarker) [inline api/MutationRecord.json:152]:
```suggestion
          "spec_url": "https://dom.spec.whatwg.org/#ref-for-dom-mutationrecord-attributenamespace②",
```

File: api/MutationRecord.json [hunk 4]
```diff
@@ -195,6 +198,7 @@
       "nextSibling": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/MutationRecord/nextSibling",
+          "spec_url": "https://dom.spec.whatwg.org/#dom-mutationrecord-nextsibling",
```
Comment (sideshowbarker) [inline api/MutationRecord.json:201]:
```suggestion
          "spec_url": "https://dom.spec.whatwg.org/#ref-for-dom-mutationrecord-nextsibling②",
```

File: api/MutationRecord.json [hunk 5]
```diff
@@ -243,6 +247,7 @@
       "oldValue": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/MutationRecord/oldValue",
+          "spec_url": "https://dom.spec.whatwg.org/#dom-mutationrecord-oldvalue",
```
Comment (sideshowbarker) [inline api/MutationRecord.json:250]:
```suggestion
          "spec_url": "https://dom.spec.whatwg.org/#ref-for-dom-mutationrecord-oldvalue②",
```

File: api/MutationRecord.json [hunk 6]
```diff
@@ -291,6 +296,7 @@
       "previousSibling": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/MutationRecord/previousSibling",
+          "spec_url": "https://dom.spec.whatwg.org/#dom-mutationrecord-previoussibling",
```
Comment (sideshowbarker) [inline api/MutationRecord.json:299]:
```suggestion
          "spec_url": "https://dom.spec.whatwg.org/#ref-for-dom-mutationrecord-previoussibling②",
```

File: api/MutationRecord.json [hunk 7]
```diff
@@ -387,6 +394,7 @@
       "target": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/MutationRecord/target",
+          "spec_url": "https://dom.spec.whatwg.org/#dom-mutationrecord-target",
```
Comment (sideshowbarker) [inline api/MutationRecord.json:397]:
```suggestion
          "spec_url": "https://dom.spec.whatwg.org/#ref-for-dom-mutationrecord-target②",
```

File: api/MutationRecord.json [hunk 8]
```diff
@@ -435,6 +443,7 @@
       "type": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/MutationRecord/type",
+          "spec_url": "https://dom.spec.whatwg.org/#dom-mutationrecord-type",
```
Comment (sideshowbarker) [inline api/MutationRecord.json:446]:
```suggestion
          "spec_url": "https://dom.spec.whatwg.org/#ref-for-dom-mutationrecord-type②",
```

File: api/PopStateEvent.json [hunk 9]
```diff
@@ -100,6 +100,7 @@
       "state": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/PopStateEvent/state",
+          "spec_url": "https://html.spec.whatwg.org/multipage/browsing-the-web.html#dom-popstateevent-state",
```
Comment (sideshowbarker) [inline api/PopStateEvent.json:103]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/browsing-the-web.html#dom-popstateevent-state-dev",
```

File: api/Request.json [hunk 10]
```diff
@@ -849,6 +849,7 @@
       },
       "isHistoryNavigation": {
         "__compat": {
+          "spec_url": "https://fetch.spec.whatwg.org/#dom-request-ishistorynavigation",
```
Comment (sideshowbarker) [inline api/Request.json:852]:
```suggestion
          "spec_url": "https://fetch.spec.whatwg.org/#ref-for-dom-request-ishistorynavigation①",
```

File: api/StorageEvent.json [hunk 11]
```diff
@@ -209,6 +210,7 @@
       "newValue": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/StorageEvent/newValue",
+          "spec_url": "https://html.spec.whatwg.org/multipage/webstorage.html#dom-storageevent-newvalue",
```
Comment (sideshowbarker) [inline api/StorageEvent.json:213]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/webstorage.html#dom-storageevent-newvalue-dev",
```

File: api/StorageEvent.json [hunk 12]
```diff
@@ -260,6 +262,7 @@
       "oldValue": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/StorageEvent/oldValue",
+          "spec_url": "https://html.spec.whatwg.org/multipage/webstorage.html#dom-storageevent-oldvalue",
```
Comment (sideshowbarker) [inline api/StorageEvent.json:265]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/webstorage.html#dom-storageevent-oldvalue-dev",
```

File: api/StorageEvent.json [hunk 13]
```diff
@@ -311,6 +314,7 @@
       "storageArea": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/StorageEvent/storageArea",
+          "spec_url": "https://html.spec.whatwg.org/multipage/webstorage.html#dom-storageevent-storagearea",
```
Comment (sideshowbarker) [inline api/StorageEvent.json:317]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/webstorage.html#dom-storageevent-storagearea-dev",
```

File: api/TextTrack.json [hunk 14]
```diff
@@ -52,6 +52,7 @@
       "activeCues": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/TextTrack/activeCues",
+          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-activecues",
```
Comment (sideshowbarker) [inline api/TextTrack.json:55]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-activecues-dev",
```

File: api/TextTrack.json [hunk 15]
```diff
@@ -251,6 +254,7 @@
       "id": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/TextTrack/id",
+          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-id",
```
Comment (sideshowbarker) [inline api/TextTrack.json:257]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-id-dev",
```

File: api/TextTrack.json [hunk 16]
```diff
@@ -354,6 +359,7 @@
       "kind": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/TextTrack/kind",
+          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-kind",
```
Comment (sideshowbarker) [inline api/TextTrack.json:362]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-kind-dev",
```

File: api/TextTrack.json [hunk 17]
```diff
@@ -402,6 +408,7 @@
       "label": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/TextTrack/label",
+          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-label",
```
Comment (sideshowbarker) [inline api/TextTrack.json:411]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-label-dev",
```

File: api/TextTrack.json [hunk 18]
```diff
@@ -450,6 +457,7 @@
       "language": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/TextTrack/language",
+          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-language",
```
Comment (sideshowbarker) [inline api/TextTrack.json:460]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-language-dev",
```

File: api/ValidityState.json [hunk 19]
```diff
@@ -99,6 +99,7 @@
       },
       "customError": {
         "__compat": {
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-validitystate-customerror",
```
Comment (sideshowbarker) [inline api/ValidityState.json:102]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-validitystate-customerror-dev",
```

File: api/ValidityState.json [hunk 20]
```diff
@@ -491,6 +492,7 @@
       },
       "valid": {
         "__compat": {
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-validitystate-valid",
```
Comment (sideshowbarker) [inline api/ValidityState.json:495]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-validitystate-valid-dev",
```

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**